lib/client-assets.php

```diff
@@ -1147,7 +1147,18 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 	// Initialize the editor.
 	$gutenberg_theme_support = get_theme_support( 'gutenberg' );
 	$align_wide              = get_theme_support( 'align-wide' );
-	$color_palette           = get_theme_support( 'editor-color-palette' );
+	$color_palette           = current( (array) get_theme_support( 'editor-color-palette' ) );
+
+	// Backcompat for Color Palette set as multiple parameters.
+	if ( is_string( $color_palette ) || isset( $color_palette['color'] ) ) {
+		$color_palette = get_theme_support( 'editor-color-palette' );
+		wp_add_inline_script(
```

Comment:
Is inlining a console.warn the way how we notice about depreciations? I think using `_doing_it_wrong()` is a safer approach.

This is probably from line 1171, but I would love to hear what @gziolo thinks about that.

Reply:
The backcompat was actually [suggested by](https://github.com/WordPress/gutenberg/pull/6437#discussion_r186741194) @jorgefilipecosta so I've left it unchanged.

Should I remove the `wp_add_inline_script` part and introduce the `_doing_it_wrong()` for both instances of console.warn?

Reply:
At the time of my suggestion _doing_it_wrong was not yet available.
I think we can now use the following:
_doing_it_wrong(
			'add_theme_support()',
			__( 'Setting colors using multiple parameters is deprecated. Please pass a single parameter with an array of colors. See https://wordpress.org/gutenberg/handbook/extensibility/theme-support/ for details.', 'gutenberg' ),
			'3.2.0'
		);

Reply:
@Luehrsen and @jorgefilipecosta I've went ahead and applied the `_doing_it_wrong` on both `console.warn` instances.